lib/incito-browser/incito.ts

```diff
@@ -827,30 +827,22 @@ export default class Incito extends MicroEvent {
         );
         this.videoObserver = new IntersectionObserver(
             (entries) => {
-                entries.forEach((entry) => {
-                    if (entry.isIntersecting) {
-                        // @ts-expect-error
-                        const autoplayState = entry.target.dataset.autoplayState;
-
-                        this.loadEl(entry.target);
-                        this.videoObserver.unobserve(entry.target);
-
-                        if (!autoplayState || autoplayState === 'paused') {
-                            // @ts-expect-error
-                            entry.target.dataset.autoplayState = 'playing';
-                            // @ts-expect-error
+                entries.forEach(async (entry) => {
+                    if (entry.target instanceof HTMLVideoElement) {
+                        if (entry.isIntersecting) {
                             entry.target.play();
+                        } else {
+                            // If loading is ongoing, we _have to_ wait for play() to be ready before pausing.
+                            if (entry.target.networkState === 2) {
+                                await entry.target.play();
+                            }
+
+                            entry.target.pause();
```

Comment:
note that redundant calls to pause and play are idempotent, i.e. if already playing(or loading) then `play()` is a no-op and if already paused `pause()` is a no-op